main.go

```diff
@@ -0,0 +1,111 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/caarlos0/env"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"time"
+)
+
+type config struct {
+	GithubToken string `env:"GHTOKEN"`
+	Authors     string `env:"AUTHORS"`
+}
+
+type UserData struct {
+	PrCount   int
+	AvatarURL string
+}
+
+type avatarResult struct {
+	AvatarURL string `json:"avatar_url"`
+}
+
+type prCountResult struct {
+	PrCount int `json:"total_count"`
+}
+
+func leaderboard(writer http.ResponseWriter, request *http.Request) {
+	userData := make(map[string]UserData)
+	authors := strings.Split(cfg.Authors, ":")
+	fmt.Printf("Authors: %v\n", authors)
+	for _, author := range authors {
+		userData[author] = UserData{PrCount: getPrCount(author), AvatarURL: getAvatar(author)}
+		fmt.Printf("Author: %s, PR count: %d\n", author, userData[author].PrCount)
+	}
+	jsonString, _ := json.Marshal(userData)
+	fmt.Fprintf(writer, "%s", jsonString)
+}
+
+func getAvatar(author string) string {
+	url := fmt.Sprintf("https://api.github.com/users/%s", author)
+	response, err := http.Get(url)
+	if response != nil {
+		defer response.Body.Close()
+	}
+	if err != nil {
+		fmt.Println("Failed to fetch avatar. %s\n", err)
+		return ""
+	} else {
+		ghData, _ := ioutil.ReadAll(response.Body)
+		result := avatarResult{}
+		json.Unmarshal([]byte(ghData), &result)
+		return result.AvatarURL
+	}
+}
+
+func getPrCount(author string) (prCount int) {
+	defer func() {
+		if r := recover(); r != nil {
```

Comment:
Is this still needed?

Reply:
Looks like it isn't, but errors if the data is missing if the auth token is wrong can't be differentiated from 0 counts.

I guess that's ok for now.